rclcpp/src/rclcpp/dynamic_subscription.cpp

```diff
@@ -41,10 +41,17 @@ void DynamicSubscription::handle_message(std::shared_ptr<void> &, const rclcpp::
 }
 
 void DynamicSubscription::handle_serialized_message(
-  const std::shared_ptr<rclcpp::SerializedMessage> &, const rclcpp::MessageInfo &)
+  const std::shared_ptr<rclcpp::SerializedMessage> & message,
+  const rclcpp::MessageInfo & message_info)
 {
-  throw rclcpp::exceptions::UnimplementedError(
-          "handle_serialized_message is not implemented for DynamicSubscription");
+  using PlainCallback = std::function<void (std::shared_ptr<SerializedMessage>)>;
```

Comment:
Maybe, to avoid duplication of the signature, we should define these before the variant, and then use them in the variant as well as here.

Reply:
Yeah that's a little less wordy, done